controllers/role.go

```diff
@@ -101,8 +102,55 @@ func (c *ApiController) UpdateRole() {
 		return
 	}
 	c.ValidateOrganization(roleFromDb.Owner)
-	
-	c.Data["json"] = wrapActionResponse(object.UpdateRole(request.Id, &role))
+
+	affected, err := object.UpdateRole(request.Id, &role)
+
+	ctx := c.getRequestCtx()
+
+	if err != nil {
+		logger.Error(ctx, "failed to update role",
+			"old_role", roleFromDb,
+			"new_role", role,
+			"error", err.Error())
+	} else if !affected {
+		logger.Error(ctx, "failed to update role: not affected",
+			"old_role", roleFromDb,
+			"new_role", role)
+	} else {
+		logger.Info(ctx, "role updated successfully",
+			"role_id", role.GetId())
+
+		oldUsers := make(map[string]struct{})
+		newUsers := make(map[string]struct{})
+
+		for _, userID := range roleFromDb.Users {
+			oldUsers[userID] = struct{}{}
+		}
+
+		for _, userID := range role.Users {
+			newUsers[userID] = struct{}{}
+		}
+
+		for userID := range oldUsers {
+			if _, found := newUsers[userID]; !found {
+				logger.Info(ctx, "user lost role",
```

Comment:
I believe we can make this message more clear to something like:"Role removed from user". It seems like the user did nothing to lose the role, so he is not an actor of this event.

Reply:
done